src/services/bookingService.ts

```diff
@@ -0,0 +1,415 @@
+        region.toLowerCase()
+    )}`;
+const generatePostUrl = (region: string) =>
+    `https://bokapass.nemoq.se/Booking/Booking/Next/${replaceSpecialChars(
+        region.toLowerCase()
+    )}`;
+const generatePreviousUrl = (region: string) =>
+    `https://bokapass.nemoq.se/Booking/Booking/Previous/${replaceSpecialChars(
+        region.toLowerCase()
+    )}`;
+
+type CheerioError = Error & { page: CheerioAPI };
+
+const confirm_booking = "Bekräfta bokning";
+
+export class BookingService {
+    fetch;
+
+    constructor(
+        public region: Location,
+        public numberOfPeople = 1,
+        public mock = false
+    ) {
+        this.region = region;
+        this.mock = mock;
+        this.fetch = (url: string, options: any = {}) => {
+            options.headers = {
+                "User-Agent":
+                    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4844.52 Safari/537.36",
+                ...options.headers,
+            };
+
+            return fetchCookie(nodeFetch)(url, options);
+        };
+
+        this.numberOfPeople = numberOfPeople;
+    }
+
+    get baseUrl() {
+        return generateBaseUrl(this.region);
+    }
+
+    get postUrl() {
+        return generatePostUrl(this.region);
+    }
+
+    async init() {
+        logger.info("Starting booking session...");
+        await this.fetch(this.baseUrl);
+
+        await this.postRequest({
+            FormId: 1,
+            ServiceGroupId: Locations[this.region].id,
+            StartNextButton: "Boka ny tid",
+        });
+        logger.success(
+            `Started booking session for ${this.numberOfPeople} person(s)`
+        );
+
+        const verifiedToken = await CaptchaService.getNewVerifiedToken();
+
+        logger.debug("Accepting booking terms...");
+        await this.postRequest({
+            AgreementText: "123",
+            AcceptInformationStorage: true,
+            NumberOfPeople: this.numberOfPeople,
+            "mtcaptcha-verifiedtoken": verifiedToken,
+            Next: "Nästa",
+        });
+        logger.success("Accepted booking terms");
+
+        logger.debug("Setting residency...");
+        await this.postRequest(
+            // eslint-disable-next-line unicorn/no-new-array
+            [...new Array(this.numberOfPeople)]
+                .map((_, index) => ({
+                    [`ServiceCategoryCustomers[${index}].CustomerIndex`]: index,
+                    [`ServiceCategoryCustomers[${index}].ServiceCategoryId`]: 2,
+                }))
+                // eslint-disable-next-line unicorn/no-array-reduce
+                .reduce(
+                    (previous, current) => ({
+                        ...previous,
+                        ...current,
+                    }),
+                    { Next: "Nästa" }
+                )
+        );
+        logger.success("Residency set");
+    }
+
+    // Can fix type later
+    async postRequest(body: any) {
```

Comment:
Could type this as `Record<string, string | number | boolean | undefined>` or something, better than nothing